src/EditorFeatures/Core.Wpf/Preview/LinesIndicator.cs

```diff
@@ -0,0 +1,174 @@
+                        }
+                    }
+
+                    // No subsequent line means we are on the last ellipsis (and do not need to do anything)
+                    UIElement adornment;
+                    if (startLine != int.MaxValue)
+                    {
+                        // Find the last line in the left source snapshot before the next ellipsis.
+                        //  Between ellipsis, there's a 1:1 correspondance between lines in the leftProjectionSnapshot and the leftSourceSnapshot so simply count lines.
+                        for (var i = leftSnapshotLine.LineNumber + 1; i < leftSnapshotLine.Snapshot.LineCount; ++i)
+                        {
+                            var nextExtent = leftSnapshotLine.Snapshot.GetLineFromLineNumber(i).Extent;
+                            if ((nextExtent.Length == PreviewFactoryService.Ellipsis.Length) && (nextExtent.GetText() == PreviewFactoryService.Ellipsis))
+                            {
+                                break;
+                            }
+
+                            ++endLine;
+                        }
+
+                        var properties = _classificationFormatMap.GetTextProperties(_lineNumberClassificaiton);
+                        var block = new TextBlock
+                        {
+                            Text = string.Format(EditorFeaturesWpfResources.LinesIndicator, startLine + 1, endLine + 1), // Displayed line numbers are 1-based.
+                            FontSize = properties.FontRenderingEmSize,
+                            FontFamily = properties.Typeface.FontFamily,
+                            Foreground = properties.ForegroundBrush,
+                            Background = _view.Background,
+                            Height = _view.LineHeight,
+                        };
+
+                        var separatorProperties = _editorFormatMap.GetProperties("outlining.verticalrule");
+                        var separatorBrush = (separatorProperties[EditorFormatDefinition.ForegroundBrushId] as Brush) ?? Brushes.Gray;
+                        var separatorY = Math.Floor(block.Height * 0.5) + 0.5;
+                        var separator = new Line()
+                        {
+                            X1 = 5.0,
+                            Y1 = separatorY,
+                            X2 = _view.ViewportWidth,
+                            Y2 = separatorY,
+                            StrokeThickness = 1.0,
+                            Stroke = separatorBrush,
+                        };
+
+                        var panel = new StackPanel() { Orientation = Orientation.Horizontal };
+                        panel.Children.Add(block);
+                        panel.Children.Add(separator);
+
+                        adornment = panel;
+                    }
+                    else
+                    {
+                        // We are on the last ellipsis and need to hide it.
```

Comment:
Why not just have the preview service not emit the ellipsis rather than having to hide it this way?